### ISSUE TITLE
Compatibility problem with rpy2

### ISSUE THREAD

**Issue by nspies** [numpy/numpy#6622]
numpy appears to be overwriting the C-level ``R_Home`` from R when it gets imported, affecting rpy2 compatibility. For example:

```python
>>> from rpy2 import robjects as ro
>>> import numpy
>>> ro.r.pdf("test2.pdf")
/Users/nspies/test/testrpy2/lib/python2.7/site-packages/rpy2/robjects/functions.py:106: UserWarning: Error in (function (file = ifelse(onefile, "Rplots.pdf", "Rplot%03d.pdf"),  : 
  failed to load default encoding

  res = super(Function, self).__call__(*new_args, **new_kwargs)
/Users/nspies/test/testrpy2/lib/python2.7/site-packages/rpy2/robjects/functions.py:106: UserWarning: In addition: 
  res = super(Function, self).__call__(*new_args, **new_kwargs)
/Users/nspies/test/testrpy2/lib/python2.7/site-packages/rpy2/robjects/functions.py:106: UserWarning: Warning message:

  res = super(Function, self).__call__(*new_args, **new_kwargs)
/Users/nspies/test/testrpy2/lib/python2.7/site-packages/rpy2/robjects/functions.py:106: UserWarning: In (function (file = ifelse(onefile, "Rplots.pdf", "Rplot%03d.pdf"),  :
  res = super(Function, self).__call__(*new_args, **new_kwargs)
/Users/nspies/test/testrpy2/lib/python2.7/site-packages/rpy2/robjects/functions.py:106: UserWarning: 

  res = super(Function, self).__call__(*new_args, **new_kwargs)
/Users/nspies/test/testrpy2/lib/python2.7/site-packages/rpy2/robjects/functions.py:106: UserWarning:  failed to load encoding file 'ISOLatin1.enc'

  res = super(Function, self).__call__(*new_args, **new_kwargs)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/nspies/test/testrpy2/lib/python2.7/site-packages/rpy2/robjects/functions.py", line 178, in __call__
    return super(SignatureTranslatedFunction, self).__call__(*args, **kwargs)
  File "/Users/nspies/test/testrpy2/lib/python2.7/site-packages/rpy2/robjects/functions.py", line 106, in __call__
    res = super(Function, self).__call__(*new_args, **new_kwargs)
rpy2.rinterface.RRuntimeError: Error in (function (file = ifelse(onefile, "Rplots.pdf", "Rplot%03d.pdf"),  : 
  failed to load default encoding
```

Importing numpy first and then importing rpy2 does not result in the error.

I've reported this on rpy2, and the author seems to think this is likely numpy's fault:

https://bitbucket.org/rpy2/rpy2/issues/311/compatibility-problem-with-numpy-and-pdf

Version info:

```python
>>> print('Python version: %s' % sys.version)
Python version: 2.7.9 (default, Dec 30 2014, 18:33:32) 
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.56)]
>>> 
>>> import numpy
>>> print('numpy path: %s' % numpy.__path__)
numpy path: ['/Users/nspies/test/testrpy2/lib/python2.7/site-packages/numpy']
>>> 
>>> import rpy2
>>> print('rpy2 version: %s' % str(rpy2.__version__))
rpy2 version: 2.8.0dev
>>> 
>>> import rpy2.rinterface
>>> print('Build against R version: %s' % str(rpy2.rinterface.R_VERSION_BUILD))
Build against R version: ('3', '2.2', '', 69053L)
>>> numpy.__version__
'1.10.1'
```

**Comment by juliantaylor** [comment]
seems to be related to the openblas affinity env update (numpy/core/__init__.py) weird

**Comment by juliantaylor** [comment]
looks like a bug or at non-defensive coding issue in R, it gets the variable from the environment but does not copy it and stores its pointer, so when its reset the pointer is invalid

**Comment by juliantaylor** [comment]
should be fixed in numpy 1.10.x

**Comment by lgautier** [comment]
Impressively short turnaround time. Thanks.
